### PR TITLE
remove unused conditional non-training codepath

### PR DESCRIPTION
**Pull request by maharajamihir**

### CODE REVIEW

**Comment by emergenz** [comment]
Just for future reference, we are hereby (implicitly) fixing a bug which was that the validation loss was calculated using the full mask instead of the actual mask.

**Review by emergenz — APPROVED** [review]
